test/functional/tests/fault_injection/test_fault_injection_core_in_raid.py

```diff
@@ -0,0 +1,70 @@
+#
+# Copyright(c) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+import re
+import pytest
+from api.cas import casadm, cli_messages
+from api.cas.cache_config import CacheMode
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet
+from storage_devices.raid import Raid, RaidConfiguration, MetadataVariant, Level
+from test_utils.size import Size, Unit
+from api.cas.cli_messages import (
+    partition_not_suitable_for_array,
+    device_or_resource_busy
+)
+
+expected_msg_1 = partition_not_suitable_for_array[0]
+expected_msg_2 = device_or_resource_busy[0]
+
+
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeSet([DiskType.sata, DiskType.hdd]))
+@pytest.mark.require_disk("core2", DiskTypeSet([DiskType.sata, DiskType.hdd]))
+def test_fault_injection_core_in_raid(cache_mode):
+    """
+        title: Test if OpenCAS rejects using core device to build SW RAID.
+        description: |
+          Test if OpenCAS handles properly attempting of use core device to build SW RAID.
+        pass_criteria:
+          - Expected to reject RAID creation with proper warning.
```

Comment:
The description suggests adding core device as a raid member whereas the test is creating the raid array with the underlaying disk. How about:
`title: Try to create raid on device used as a core device`
`description: Verify that it is impossible to use an underlaying core disk as raid member`